androidlibrary_lib/src/main/java/org/opendatakit/properties/PropertiesSingleton.java

```diff
@@ -488,6 +488,7 @@ private void init() {
     // and update them from the mDeviceDefaults map.
     for (Map.Entry<String, String> entry : mDeviceDefaults.entrySet()) {
       if (!mDeviceProps.containsKey(entry.getKey())) {
+        if(entry.getKey() != null && entry.getValue() != null)
```

Comment:
You should have brackets around 491, 492 and 493. As you don't want to set the updatedDeviceProps flag to true unless you actually set a property.